include/swift/Runtime/BuiltinProtocolConformances.h

```diff
@@ -0,0 +1,82 @@
+//===--- BuiltinProtocolWitnessTable.h --------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Swift runtime support for builtin protocol witnesses and related items.
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/ABI/Metadata.h"
+
+namespace swift {
+
+// public protocol Equatable {}
+#define SWIFT_EQUATABLE_MANGLING SQ
+
+// == infix(T, T) -> Swift.Bool
+#define SWIFT_EQUAL_OPERATOR_MANGLING 2eeoiySbx_xtFZ
+
+// (A...)
+#define VARIADIC_TUPLE_MANGLING xd_t
```

Comment:
For now, would this mangling only be used for the witness symbols?

Reply:
Yes, this mangling would only be used for the witness symbols. I imagine variadic generics would get their own mangling separate from a variadic generic type, but if they aren't then this might pose an issue.

Reply:
As long as it's only for the symbols, I think we're OK, as long as we note that this mangling sequence is used for that purpose. It's not hard to hard code existing symbol names of things for ABI stability purposes if we do establish a more generic variadic mangling in the future.

Another possibility might be to just give these symbols fixed unmangled names like `_swift_tupleEquatableWitnesses`, `_swift_tupleEquatable_equals`, etc. that the compiler knows.

Reply:
I've gone ahead and just used the unmangled names to play it safe just in case that mangling is ever needed in the future.